### PR TITLE
Send what features we support to the server.

### PR DESCRIPTION
**Pull request by LoneBoco**
Some servers will listen to what features a client says it supports and will disable those features, returning "False" in the server capabilities reply.  Which, in turn, disables that feature in our code.

### CODE REVIEW

**Comment by LoneBoco** [comment]
The server in question where I encountered this issue was https://github.com/OmniSharp/omnisharp-roslyn

I'm not sure if sending everything in completionItemKind is correct or not.  I figured I may as well.

File: plugin/core/main.py [hunk 1]
```diff
@@ -217,17 +217,80 @@ def start_client(window: sublime.Window, config: ClientConfig):
...
+                            CompletionItemKind.Reference
+                            ]
                     }
                 },
-                "synchronization": {
-                    "didSave": True
-                }
+                "signatureHelp": {
+                    "signatureInformation": {
+                        "documentationFormat": ["plaintext"]
```
Comment (rwols) [inline plugin/core/main.py]: I'm fairly certain you can set this to `markdown`.

Comment (LoneBoco) [reply plugin/core/main.py]: I've added markdown to the list.

**Comment by rwols** [comment]
I’m eager to try this out tomorrow!

**Comment by tomv564** [comment]
Whoops, I see the URI fix is also in here, could you keep that in the other PR?

**Comment by LoneBoco** [comment]
@tomv564 I thought I reverted my accidental merge.  Is it still showing?

**Comment by tomv564** [comment]
It's likely I only saw the old commit and didn't check the diff - sorry to bother you.
Thanks for the initiative, and good to hear omnisharp being used with LSP!

**Comment by rwols** [comment]
@LoneBoco do you happen to work with macOS? I tried several times to install the OmniSharp LSP server but never got it to work.

**Comment by LoneBoco** [comment]
@rwols I do not.  I use [omnisharp-roslyn](https://github.com/OmniSharp/omnisharp-roslyn) in Windows.  According to the README.md, you will need mono (>=5.2.0) to run it on OSX.  You should be able to clone it and run `build.sh` to have it download everything, build it, and run all the tests.

These are the settings I use with LSP:
```
{
    "clients":
    {
        "csharp":
        {
            "command": ["C:/Projects/omnisharp-roslyn/artifacts/scripts/OmniSharp.Stdio.cmd", "-lsp"],
            "scopes": ["source.cs"],
            "syntaxes": ["Packages/C#/C#.sublime-syntax"],
            "languageId": "csharp"
        }
    }
}
```

One thing to note is that omnisharp-roslyn has issues with the current master branch.  Some of them are partially fixed, but haven't been merged back yet.

I've [forked omnisharp-roslyn](https://github.com/LoneBoco/omnisharp-roslyn) and pushed some of the fixes into my repository, as well as added my own fixes for https://github.com/OmniSharp/omnisharp-roslyn/issues/1113, https://github.com/OmniSharp/omnisharp-roslyn/issues/1119, and https://github.com/OmniSharp/omnisharp-roslyn/issues/1125.  If you want the most problem free version of omnisharp-roslyn to mess with, I would recommend you clone from me for the time being.

Also, I would recommend you run with logging enabled.  I had originally tried to open a project and it didn't work.  The logs showed that msbuild was not able to process the solution as the project was using .NET 4.7 and I didn't have that version installed on my machine.  That might be a reason why you are having issues.

**Pull request merged**